File: FootballExplorer/app/src/main/java/fr/a42consulting/footballexplorer/Model/TeamsSearchTeam.java
package fr.a42consulting.footballexplorer.Model;

import org.json.JSONException;
import org.json.JSONObject;

public class TeamsSearchTeam {
    public static TeamsSearchTeam fromJson(JSONObject jsonObject) {
        TeamsSearchTeam team = new TeamsSearchTeam();
        try {
            team.id = jsonObject.getString("idTeam");
            team.name = jsonObject.getString("strTeam");
            team.imageUrl = jsonObject.getString("strTeamBadge");
        } catch (JSONException e) {
            e.printStackTrace();
            return null;
        }
        return team;
    }

    private String id;
    private String name;
    private String imageUrl;

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getImageUrl() {
        return imageUrl;
    }
}

File: FootballExplorer_kotlin/app/src/main/java/fr/a42consulting/footballexplorer/model/TeamsSearchTeam.kt
package fr.a42consulting.footballexplorer.model

import org.json.JSONException
import org.json.JSONObject

class TeamsSearchTeam {

    lateinit var id: String
        private set
    lateinit var name: String
        private set
    lateinit var imageUrl: String
        private set

    companion object {
        fun fromJson(jsonObject: JSONObject): TeamsSearchTeam? {
            val team = TeamsSearchTeam()
            try {
                team.id = jsonObject.getString("idTeam")
                team.name = jsonObject.getString("strTeam")
                team.imageUrl = jsonObject.getString("strTeamBadge")
            } catch (e: JSONException) {
                e.printStackTrace()
                return null
            }

            return team
        }
    }
}
